pkg/cmd/server/kubernetes/master.go

```diff
@@ -11,6 +11,7 @@ import (
 
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/record"
+	osclient "github.com/openshift/origin/pkg/client"
```

Comment:
> So each controller would have to loop through all HPAs, run the check to see if the given scale resource pointed at Origin or Kube, and then skip it if it pointed at the wrong resource type? That seems a bit redundant...

There's no other correct way to handle it.  A given HPA controller should only attempt to handle scale requests for kinds that it can understand.  It gets back to my points about being able to recognize which resources a scaler client is good for.